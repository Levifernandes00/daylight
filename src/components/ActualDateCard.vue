<template>
  <main class="bg-white shadow-1 border-round flex-grow-1 p-2 m-1">
    <div class="px-6 py-3">
      <div class="py-1 text-2xl">
        <strong>Actual Date</strong>
      </div>
      <div class="py-2">{{ today }}</div>
    </div>
  </main>
</template>

<script lang="ts">
import { DateTime } from 'luxon'
import { defineComponent } from 'vue'

export default defineComponent({
  data() {
    return {
      today: ''
    }
  },
  created() {
    this.setToday()
  },
  methods: {
    setToday() {
      this.today = DateTime.now()
        .setLocale('it-IT')
        .toLocaleString(DateTime.DATE_MED_WITH_WEEKDAY)
    }
  }
})
</script>

<style lang="scss"></style>
