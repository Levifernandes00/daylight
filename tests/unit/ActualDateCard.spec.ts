import ActualDateCard from '@/components/ActualDateCard.vue'
import { flushPromises, shallowMount } from '@vue/test-utils'
import { DateTime } from 'luxon'

describe('ActualDateCard.vue ', () => {
  it('matches snapshot',async () => {
    const wrapper = shallowMount(ActualDateCard)
    wrapper.setData({today: DateTime.utc(2022, 9, 22)})
    await flushPromises()
    expect(wrapper).toMatchSnapshot()
  })

  it('renders actual date title', () => {
    const wrapper = shallowMount(ActualDateCard)
    expect(wrapper.get('strong').text()).toMatch('Actual Date')
  })

  it('renders proper day', () => {
    const expectedDate = DateTime.now().setLocale('it-IT').toLocaleString({
      weekday: 'short',
      day: 'numeric',
      month: 'short',
      year: 'numeric'
    })
    const wrapper = shallowMount(ActualDateCard)
    expect(wrapper.text()).toContain(expectedDate)
  })
})
