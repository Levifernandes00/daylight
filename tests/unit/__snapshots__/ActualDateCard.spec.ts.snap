// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActualDateCard.vue  matches snapshot 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <main
        class="bg-white shadow-1 border-round flex-grow-1 p-2 m-1"
      >
        <div
          class="px-6 py-3"
        >
          <div
            class="py-1 text-2xl"
          >
            <strong>
              Actual Date
            </strong>
          </div>
          <div
            class="py-2"
          >
            2022-09-22T00:00:00.000Z
          </div>
        </div>
      </main>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {},
      "da": null,
      "data": Object {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {},
      "refs": Object {
        "VTU_COMPONENT": Object {
          "hasOwnProperty": [Function],
          "setToday": [Function],
          "today": "2022-09-22T00:00:00.000Z",
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": Object {},
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "hasOwnProperty": 4,
            "isVueInstance": 0,
            "setToday": 4,
            "toJSON": 0,
            "today": 2,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "hasOwnProperty": [Function],
            "setToday": [Function],
            "today": "2022-09-22T00:00:00.000Z",
          },
          "da": null,
          "data": Object {
            "today": "2022-09-22T00:00:00.000Z",
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {},
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {},
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "hasOwnProperty": [Function],
            "setToday": [Function],
            "today": "2022-09-22T00:00:00.000Z",
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": Array [
                      Object {
                        "__v_isVNode": true,
                        "__v_skip": true,
                        "anchor": null,
                        "appContext": null,
                        "children": "Actual Date",
                        "component": null,
                        "dirs": null,
                        "dynamicChildren": null,
                        "dynamicProps": null,
                        "el": <strong>
                          Actual Date
                        </strong>,
                        "key": null,
                        "patchFlag": 0,
                        "props": null,
                        "ref": null,
                        "scopeId": null,
                        "shapeFlag": 9,
                        "slotScopeIds": null,
                        "ssContent": null,
                        "ssFallback": null,
                        "staticCount": 0,
                        "suspense": null,
                        "target": null,
                        "targetAnchor": null,
                        "transition": null,
                        "type": "strong",
                      },
                    ],
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <div
                      class="py-1 text-2xl"
                    >
                      <strong>
                        Actual Date
                      </strong>
                    </div>,
                    "key": null,
                    "patchFlag": -1,
                    "props": Object {
                      "class": "py-1 text-2xl",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 17,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "div",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "2022-09-22T00:00:00.000Z",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <div
                      class="py-2"
                    >
                      2022-09-22T00:00:00.000Z
                    </div>,
                    "key": null,
                    "patchFlag": 1,
                    "props": Object {
                      "class": "py-2",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "div",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": null,
                "key": null,
                "patchFlag": 0,
                "props": Object {
                  "class": "px-6 py-3",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "2022-09-22T00:00:00.000Z",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="py-2"
                >
                  2022-09-22T00:00:00.000Z
                </div>,
                "key": null,
                "patchFlag": 1,
                "props": Object {
                  "class": "py-2",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "dynamicProps": null,
            "el": <main
              class="bg-white shadow-1 border-round flex-grow-1 p-2 m-1"
            >
              <div
                class="px-6 py-3"
              >
                <div
                  class="py-1 text-2xl"
                >
                  <strong>
                    Actual Date
                  </strong>
                </div>
                <div
                  class="py-2"
                >
                  2022-09-22T00:00:00.000Z
                </div>
              </div>
            </main>,
            "key": null,
            "patchFlag": 0,
            "props": Object {
              "class": "bg-white shadow-1 border-round flex-grow-1 p-2 m-1",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "main",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "created": [Function],
            "data": [Function],
            "methods": Object {
              "setToday": [Function],
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <main
          class="bg-white shadow-1 border-round flex-grow-1 p-2 m-1"
        >
          <div
            class="px-6 py-3"
          >
            <div
              class="py-1 text-2xl"
            >
              <strong>
                Actual Date
              </strong>
            </div>
            <div
              class="py-2"
            >
              2022-09-22T00:00:00.000Z
            </div>
          </div>
        </main>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "created": [Function],
          "data": [Function],
          "methods": Object {
            "setToday": [Function],
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <main
          class="bg-white shadow-1 border-round flex-grow-1 p-2 m-1"
        >
          <div
            class="px-6 py-3"
          >
            <div
              class="py-1 text-2xl"
            >
              <strong>
                Actual Date
              </strong>
            </div>
            <div
              class="py-2"
            >
              2022-09-22T00:00:00.000Z
            </div>
          </div>
        </main>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.39",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "hasOwnProperty": [Function],
    "setToday": [Function],
    "today": "2022-09-22T00:00:00.000Z",
  },
  "isDisabled": [Function],
  "rootVM": Object {},
  "wrapperElement": <main
    class="bg-white shadow-1 border-round flex-grow-1 p-2 m-1"
  >
    <div
      class="px-6 py-3"
    >
      <div
        class="py-1 text-2xl"
      >
        <strong>
          Actual Date
        </strong>
      </div>
      <div
        class="py-2"
      >
        2022-09-22T00:00:00.000Z
      </div>
    </div>
  </main>,
}
`;
